src/lib.rs

```diff
@@ -242,47 +316,113 @@ impl StaticTypeMap {
             .and_then(|any| any.downcast().ok())
             .map(|concrete_type| *concrete_type)
     }
+}
 
-    /// Reserves capacity for at least `additional` more types to be inserted in the map. The
-    /// collection may reserve more space to avoid frequent reallocations.
+impl GenericStaticTypeMap<dyn Any + Send> {
```

Comment:
This could be a macro, e.g:
```rust
impl_bounded_map!(Any, Any + Send, /* etc */);
```

My previous comment about macros doesn't mean you shouldn't use them, they are really useful when used in the right context.

Edit: I'm here if you need help writing the macro.